### PR TITLE
HDDS-4737. Add ModifierOrder to checkstyle rules

### PR DESCRIPTION
**Pull request by lamber-ken**
## What changes were proposed in this pull request?

Add ModifierOrder to checkstyle rules
- `public final static` to `public static final`
- `static public` to `public static`
- etc

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4737

## How was this patch tested?

No need new tests.

### CODE REVIEW

**Comment by amaliujia** [comment]
+1 seems be fine to enforce modifier order from now. The existing violation thus fix in this PR is not too large.

**Comment by lamber-ken** [comment]
> +1 seems be fine to enforce modifier order from now. The existing violation thus fix in this PR is not too large.

Thanks @amaliujia

**Comment by ayushtkn** [comment]
Well this looks good to have, but this is changing the rules at the project level, I think such changes if required should be notified on the dev list.

FYI. Hadoop too had this checkstyle rule, but was disabled as part of [HADOOP-12713](https://issues.apache.org/jira/browse/HADOOP-12713)

**Comment by amaliujia** [comment]
Sounds reasonable to at least mention this in dev list as it will impact all developers (though the impact is supposed to be minor)

**Comment by lamber-ken** [comment]
Got @amaliujia @ayushtkn

**Comment by lamber-ken** [comment]
hi @amaliujia @ayushtkn @elek, please review this pr again if you are free, thanks 😄

**Comment by lamber-ken** [comment]
hi @amaliujia @ayushtkn @elek, please review this pr again if you are free, thanks. If the merge operation is late, it will cause more conflicts.

**Comment by lamber-ken** [comment]
hi @adoroszlai , please review this pr again if you are free, thanks. If the merge operation is late, it will cause more conflicts.

**Review by ayushtkn — APPROVED** [review]
There is no objection on the [MailingList](https://lists.apache.org/thread.html/ra82429d3ae977ec1b5b0b2b194d74fba040765386751bcffd5b3e26a%40%3Cdev.ozone.apache.org%3E) 
We are good to go then. +1

**Comment by lamber-ken** [comment]
> There is no objection on the [MailingList](https://lists.apache.org/thread.html/ra82429d3ae977ec1b5b0b2b194d74fba040765386751bcffd5b3e26a%40%3Cdev.ozone.apache.org%3E)
> We are good to go then. +1

Thanks 👍

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @lamber-ken for proposing this.  The meat of the change looks good.

File: pom.xml [hunk 1]
```diff
@@ -1974,6 +1974,7 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
             <suppressionsLocation>hadoop-hdds/dev-support/checkstyle/suppressions.xml</suppressionsLocation>
             <includeTestSourceDirectory>true</includeTestSourceDirectory>
             <failOnViolation>false</failOnViolation>
+            <consoleOutput>true</consoleOutput>
```
[the remark this answers is not in the document]
Comment (lamber-ken) [reply pom.xml]: Got, updated.

**Review by adoroszlai — APPROVED** [review]
Thanks @lamber-ken for updating the patch.

**Comment by adoroszlai** [comment]
Thanks @lamber-ken for the improvement, @amaliujia and @ayushtkn for the review.

**Comment by lamber-ken** [comment]
> Thanks @lamber-ken for the improvement, @amaliujia and @ayushtkn for the review.

Welcome.

**Pull request merged**